tools/Az.Tools.Predictor/Az.Tools.Predictor/ParameterValuePredictor.cs

```diff
@@ -55,18 +60,23 @@ public void ProcessHistoryCommand(CommandAst command)
         /// <returns>The parameter value from the history command. Null if that is not available.</returns>
         public string GetParameterValueFromAzCommand(string commandNoun, string parameterName)
         {
-            if (_command_param_to_resource_map.ContainsKey(commandNoun))
+            parameterName = parameterName.ToLower();
+            var key = parameterName;
+            Dictionary<string, string> commandNounMap = null;
+
+            if (_command_param_to_resource_map?.TryGetValue(commandNoun, out commandNounMap) == true)
             {
-                parameterName = parameterName.ToLower();
-                if (_command_param_to_resource_map[commandNoun].ContainsKey(parameterName))
+                if (commandNounMap.TryGetValue(parameterName, out var parameterNameMappedValue))
                 {
-                    var key = _command_param_to_resource_map[commandNoun][parameterName];
-                    if (_localParameterValues.TryGetValue(key, out var value))
-                    {
-                        return value;
-                    }
+                    key = parameterNameMappedValue;
                 }
             }
+
+            if (_localParameterValues.TryGetValue(key, out var value))
+            {
+                return value;
+            }
+
             return null;
```

Comment:
I realize this isn't a new change for this PR, but would an empty string work here instead of a null?

Reply:
I think the caller doesn't care the case if it's null or empty string and will treat them the same. But I also think that `null` has the meaning that the value it's looking for doesn't exist v.s. it exists and it's empty. In that case, I'll keep `null` for now.